test/dlrm/test/dlrm_dot_test.hpp

```diff
@@ -55,6 +57,13 @@ namespace rocwmma
             auto problemSize   = std::get<2>(param);
             auto passDirection = std::get<3>(param);
 
+            // Cleanup previously used resources if data types change
+            if (sLastKernelRun && sLastKernelRun->getResource() != kernel->getResource())
+            {
+                sLastKernelRun->resetMemory();
```

Comment:
With proposed changes, this would turn into sLastKernelRun->getResource()->reset()